EIPS/eip-2569.md

```diff
@@ -0,0 +1,357 @@
+
+For a fungible token such as an ERC-20 token no such an ID is needed and our proposed interface just assigns an SVG image's file content to a string variable of the token's contract.
+
+```imageSvg```: we use a string variable to save an SVG image file's content.
+
+An SVG image that will be saved in the imageSvg string should include at least two attributes:"name", "desc"(description).
+
+```setTokenImageSvg```: for an ERC-721 or ERC-1155 token or a token implemented by a contract which has a member "ID" to specify its token type or token index we define an interface to associate an SVG image to the token's ID number. For an ERC-20 token or a token implemented by a contract which doesn't have a member "ID" to specify its token type or token index we define an interface to assign an SVG image to a member variable string of this token's contract.
+
+```getTokenImageSvg```: for an ERC-721 or ERC-1155 token or a token implemented by a contract which has a member "ID" to specify its token type or token index we define an interface to get an SVG image by using the token's ID number. For an ERC-20 token or a token implemented by a contract which doesn't have a member "ID" to specify its token type or token index we define an interface to get an SVG image for it if the token has a member variable string to save the image.
+
+
+## Specification
+Our proposed interfaces define how a smart contract saves an image for a universal token in Ethereum which keeps the image permanent and tamper-resistant, and how a smart contract retrieves an image from Ethereum for a universal token.  
+
+By calling our proposed interfaces we assume users are accessing an SVG image. 
+
+The procedure to save an image for a token in Ethereum is as follows:
+
+**Step1:** define a string variable or an array of strings to hold an image or an array of images.
+
+**Step 2:** define a function to set an (SVG) image's file content or an array of image file's contents to the string variable or the array of strings.
+
+The procedure to retrieve an image for a token from Ethereum is as follows:
+
+Step 1: for a token such as an ERC-721 or ERC-1155 token which has a member variable "ID"  to specify a token type or index and a member variable string to keep an (SVG) image associated with the "ID", retrieve the (SVG) image from Ethereum by calling our proposed "get" interface with the token's ID; 
+for a token which doesn't have a member variable "ID" to specify a token type of index but has a member variable string to keep an (SVG) image, retrieve the (SVG) image from Ethereum by calling our proposed "get" without an "ID". 
```

Comment:
Consider defining the interface in the specification section as a Solidity interface, and include notes on any required behavior of the functions.

Reply:
I am not sure if I get your point. Would you please elaborate this a little bit? Did you mean I move the " `setTokenImageSvg`: for an ERC-721 .....`getTokenImageSvg`: for an ERC-721 ......" from the "Definitions" section to here , after "**Step 2:**......" and before "Step 1: for a token......" ?

Reply:
A good specification may look something like this:
----
Implementations of this standard **MUST** implement the following interface:
```solidity
interface Erc2569 {
    function getTokenImageSvg(uint256 id) returns (string memory);
    function setTokenImageSvg(string data) returns (uint256 id);
}
```
If you pass the `id` returned by `setTokenImageSvg` into `getTokenImageSvg` then `getTokenImageSvg` **MUST** return the same value that `setTokenImageSvg` was provided.

Reply:
I got your point after I reviewed EIP-1. Changes made and committed